pkg/customresourcestate/registry_factory.go

```diff
@@ -503,7 +505,8 @@ func addPathLabels(obj interface{}, labels map[string]valuePath, result map[stri
 		m := labels[k].Get(obj)
 		if kv, ok := m.(map[string]interface{}); ok {
 			for k, v := range kv {
-				result[k] = fmt.Sprintf("%v", v)
+				otelCompatPromLabelName := otelprom.NormalizeLabel(k)
+				result[otelCompatPromLabelName] = fmt.Sprintf("%v", v)
```

Comment:
[sanitizeLabelName](https://github.com/kubernetes/kube-state-metrics/blob/main/internal/store/utils.go#L129) felt a bit more "fitting" to me for this usecase, so I ended up using that here. LMK if you think otherwise and I'll change this accordingly.